### PR TITLE
Get all places

### PR DESCRIPTION
**Pull request by RQ16**
#10

### CODE REVIEW

**Comment by m4v15** [comment]
@RQ16 @JWLD 
So just couple of things before you move on:

 - This is just for the `/en/` end point at the moment - the `/ar/` is going to be exactly the same but at [these lines](https://github.com/foundersandcoders/OTP-Data-Entry/pull/24/files#diff-9fd324f26492298896db69d1b1b4ab3R8) just change it to `this.ar.whatever`

- Regarding that, instead of [having text written directly in handlebars](https://github.com/foundersandcoders/OTP-Data-Entry/pull/24/files#diff-89fd324f26492298896db69d1b1b4ab3R1), we should have a javascript file which holds a languages object with both english and arabic, that we require in at the top of the server, then we can use it in handlebars. A bit like the way we did it in easyNaz:
   - [text file](https://github.com/FACN1/easyNaz/blob/master/text.js)
   - [using it](https://github.com/FACN1/easyNaz/blob/master/app.js#L17-L25)

This stuff might look quite mad and complicate so we come get me if you wanna talk through it.

File: src/controllers/router.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 const router = require('express').Router();
 
-router.get('/', require('./home.js'))
+router.get('/', require('./home.js'));
+router.get('/en/places', require('./places.js').getAll);
```
Comment (m4v15) [inline src/controllers/router.js]: I don't really like this layout,  it looks a bit confusing - think it'd be better to require `./places` at the top as `placesController` or something and then do `placesController.getAll` here

**Comment by m4v15** [comment]
Sorry, I should really be reviewing this before you assign me, just a few things I thought of ☝️ 

It all looks 👌  apart from the stuff I mentioned

**Comment by jwld** [comment]
Cheers @m4v15 , when you linked to 'these lines' it links to the general PR for me - which lines did you mean?

**Comment by m4v15** [comment]
@JWLD sorry habibi I meant here https://github.com/foundersandcoders/OTP-Data-Entry/blob/f390363f3df118f7420c7dce9e0411a386af15ae/views/places.hbs#L1

**Comment by m4v15** [comment]
Oooo wow I've never seen it do that in comment rendering before

File: src/app.js [hunk 2]
```diff
@@ -1,9 +1,12 @@
 const express = require('express');
 const router = require('./controllers/router.js');
 const hbs = require('express-handlebars');
+const languages = require('./lang.js');
 
 const app = express();
 
+app.locals.text = languages['english'];
```
Comment (m4v15) [inline src/app.js]: we're not always going to want this to be english.

Could do:

```
app.locals.textEn = languages['english']
app.locals.textAr = languages['arabic']
```

Actually, why not `languages.english`?

File: src/controllers/places.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+const Request = require('request');
+
+const placeController = {};
+
+placeController.getAll = (req, res) => {
+  const placesURL = 'https://nazareth-open-tourism-platform.herokuapp.com/places';
+  Request(placesURL, (error, response, body) => {
+    if (error) return res.status(404).send(error);
+    const data = JSON.parse(body);
```
Comment (m4v15) [inline src/controllers/places.js]: I think we need to filter the data here to make sure we only have results that contain `en.name` fields

File: views/places.hbs [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<h1 class="page-title">{{localLang.placesTitle}}</h1>
+
+<ul>
+  {{#each output}}
+    <li class="place-wrap">
+      <img class="place-wrap__image" src="{{this.imageUrl}}" alt="Place Image">
+      <div class="place-wrap__info">
+        <span class="place-wrap__span">{{en.name}}</span>
+        <span class="place-wrap__span">{{en.description}}</span>
```
Comment (m4v15) [inline views/places.hbs]: So anything that's not `required` in the API schema [here](https://github.com/foundersandcoders/open-tourism-platform/blob/master/docs.md#create-place) we need to check whether or not it exists in the response

Comment (m4v15) [reply views/places.hbs]: Otherwise we might get back a load of undefined/empty responses

File: src/controllers/places.js (hunk 3 above)
Comment (m4v15) [reply src/controllers/places.js]: using `array.filter`

**Comment by m4v15** [comment]
So is this done?

**Comment by RQ16** [comment]
Yes, @m4v15

**Comment by RQ16** [comment]
Functionality wise, but no styling

**Comment by m4v15** [comment]
Cool, take off the WIP tag and write a message so i know to review and merge or request changes. @RQ16

**Review by m4v15 — CHANGES_REQUESTED** [review]
What happened to the `dir` stuff? Did you decide against it? Apart from that this looks good.

**Comment by m4v15** [comment]
@RQ16 if we finish this before we get started on any others that would be 👌 .

So in the placesController we should find out what language we are in (using `req.params.lang`) and then have a switch statement to set a variable called `dir` as either `ltr` or `rtl` (left-to-right, etc)

We can then pass this as data into the view, and in the html tag in the layouts/main.hbs, here:
https://github.com/foundersandcoders/OTP-Data-Entry/blob/2b2389d687604bd03ed9c8567c4b3a151fbbe0b1/views/layouts/main.hbs#L2

You can add a text direction attribute, like [here](https://github.com/FACN1/easyNaz/blob/master/views/layouts/main.hbs)

**Comment by RQ16** [comment]
@m4v15 Is adding the lang variable as an attribute essential here? Is the dir one not sufficient?

**Comment by m4v15** [comment]
If we have access to it, we might has well, looks like it might be useful

https://www.w3.org/International/questions/qa-lang-why

(These are the language codes)[http://www.iana.org/assignments/language-subtag-registry/language-subtag-registry] 

So it looks like 'en' and 'ar' will work

So our controller will look something like this

```js
// Some if statement to work out and set a dir variable

res.render('places', {
       output: data,
       localLang: req.app.locals[req.params.lang].
       lang: req.params.lang
       dir
     });
```

**Comment by m4v15** [comment]
🎉 😍

**Review by m4v15 — APPROVED** [review]

**Pull request merged**